src/constants/TokenSafetyLookupTable.ts

```diff
@@ -0,0 +1,49 @@
+import store from '../state'
+import { UNI_EXTENDED_LIST, UNI_LIST } from './lists'
+import brokenTokenList from './tokenLists/broken.tokenlist.json'
+import unsupportedTokenList from './tokenLists/unsupported.tokenlist.json'
+
+export enum TOKEN_LIST_TYPES {
+  UNI_DEFAULT = 1,
+  UNI_EXTENDED,
+  UNKNOWN,
+  BLOCKED,
+  BROKEN,
+}
+
+class TokenSafetyLookupTable {
+  dict: { [key: string]: TOKEN_LIST_TYPES } = {}
+
+  uniDefaultTokens = store.getState().lists.byUrl[UNI_LIST].current?.tokens
+  brokenTokens = brokenTokenList.tokens
+  unsupportTokens = unsupportedTokenList.tokens
+  uniExtendedTokens = store.getState().lists.byUrl[UNI_EXTENDED_LIST].current?.tokens
+
+  constructor() {
+    if (!this.uniDefaultTokens) {
+      this.uniDefaultTokens = []
+    }
+    if (!this.uniExtendedTokens) {
+      this.uniExtendedTokens = []
+    }
+
+    this.brokenTokens.forEach((token) => {
```

Comment:
i worry about a future where these token lists grow large and this is a lot of startup initialization. the best idea i have around that is do make `checkToken` async and lazily initialize. wdyt?

Reply:
Yeah forcing it to run through an extra couple iterations makes it take like 3 seconds. Maybe there is a way to have redux initialize the table as its initializing itself? Or maybe redux only unpacks the default list and doesn't unpack extended... I'll look into it, but curious to get your thoughts @cmcewen